docs/docs/sourcing-from-contentstack.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Sourcing from Contentstack
+---
+
+This guide walks you through the steps involved in setting up a Gatsby site that fetches content from [Contentstack](https://www.contentstack.com/), a popular headless (or API-first) CMS.
+
+## Why use Contentstack with Gatsby?
+
+Contentstack is a [headless CMS](https://www.contentstack.com/) that offers developers turnkey Content-as-a-Service (CaaS). It lets you create flexible content components and deliver them to any destination platform via CDN-backed Content Delivery APIs. Instead of having to create separate content for all your sites and apps, Contentstack lets you reuse content across different digital channels (web, mobile, IoT, voice, etc.) and present it optimized for its digital endpoint, be it a browser on a laptop, a smartphone, a smartwatch, a kiosk, or a jumbotron.
+
+Gatsby, with its powerful UI toolset, serves as the “head” for your websites while Contentstack – being the headless CMS – acts as the “body”. Combining these the two best-in-class tools provide a powerful solution to create modern, enterprise-grade sites and apps.
+
+_**Note:** This guide uses the `gatsby-source-contentstack` plugin to create a Gatsby-powered site._
+
+## Build a Site with Contenstack and Gatsby
+
+### Prerequisites
+
+- [ ] Install Node.js on your system
+- [ ] Install Gatsby CLI: `npm install -g gatsby-cli`
+
+### Step 1: Clone the repo
+
+Clone the repo from GitHub. This contains all the required dependencies.
+
+`git clone https://github.com/contentstack/gatsby-starter-contentstack.git`
+
+### Step 2: Install npm
+
+Go to the ‘gatsby-starter-contentstack’ folder, and run the following:
+
+```cd gatsby-starter-contentstack
+npm install
+```
+
+This downloads the required files and initializes the site.
+
+### Step 3: Update the config file
+
+Add the `api_key`, `delivery_token` and `environment` stack details to the existing `gatsby-source-contentstack` object within the gatsby-config.js file:
+
+```plugins: [{
+    resolve: `gatsby-source-contentstack`,
+    options: {
+      api_key: `api_key`,
+      access_token: `delivery_token`,
```

Comment:
maybe point the user to use `.env` variables to not expose private keys in public repositories...

Reply:
Thanks @muescha, and apologies on the delay. Thanks for reviewing!